src/main/scala/uk/gov/nationalarchives/omega/api/business/agents/ListAgentSummaryService.scala

```diff
@@ -83,28 +88,41 @@ class ListAgentSummaryService(val stubData: StubData, val repository: AbstractRe
       Try(dateFormatter.parse(trimmedDate)).toOption
   }
 
-  private def getAgentSummaries(listAgentSummary: ListAgentSummary): List[AgentSummary] =
-    listAgentSummary.depository match {
-      case Some(true) => getPlaceOfDepositSummaries
-      case _          => getAgentEntities
-    }
+  private def getAgentSummaries(listAgentSummary: ListAgentSummary): Try[List[AgentSummary]] =
+    for {
+      agentEntities  <- repository.getAgentSummaryEntities(listAgentSummary.copy(versionTimestamp = None))
+      agentSummaries <- convertAgentSummaryEntities(agentEntities, listAgentSummary)
+    } yield agentSummaries
 
-  private def getPlaceOfDepositSummaries: List[AgentSummary] =
-    repository.getPlaceOfDepositEntities match {
-      case Success(agentEntities) =>
-        agentEntities.flatMap { agentEntity =>
-          agentEntity.as[Option[AgentSummary]]
-        }
-      case _ => List.empty // TODO (RW) log the error
-    }
+  private def combineSummaryAndDescriptions(
+    agentSummary: AgentSummary,
+    agentDescriptions: List[AgentDescription]
+  ): Try[AgentSummary] =
+    Try(agentSummary.copy(descriptions = agentDescriptions))
 
-  private def getAgentEntities: List[AgentSummary] =
-    repository.getAgentEntities match {
-      case Success(agentEntities) =>
-        agentEntities.flatMap { agentEntity =>
-          agentEntity.as[Option[AgentSummary]]
-        }
-      case _ => List.empty // TODO (RW) log the error
-    }
+  private def convertAgentSummaryEntities(
+    agentSummaryEntities: List[AgentSummaryEntity],
+    listAgentSummary: ListAgentSummary
+  ): Try[List[AgentSummary]] =
+    agentSummaryEntities.map { agentSummaryEntity =>
+      agentSummaryEntity.as[Option[AgentSummary]] match {
+        case Some(agentSummary) =>
+          for {
+            agentDescriptions            <- getAgentDescriptions(agentSummaryEntity, listAgentSummary)
+            agentSummaryWithDescriptions <- combineSummaryAndDescriptions(agentSummary, agentDescriptions)
+          } yield agentSummaryWithDescriptions
+        case _ => Failure(new Exception("")) // TODO(RW) what exception to use here?
```

Comment:
Still needs to be resolved perhaps?

Reply:
Have now created a `ServiceException` class for this and other similar errors.